data/data/azure/vnet/nsg.tf

```diff
@@ -1,26 +1,36 @@
 resource "azurerm_network_security_group" "master" {
+  count               = var.preexisting_network ? 0 : 1
+
```

Comment:
Do we really need and/or want to gate on security groups when byo vnet?

Reply:
Yes, reasoning as described above.